File: 3811-reverse-degree-of-a-string/reverse-degree-of-a-string.java
class Solution {
    public int reverseDegree(String s) {
        int totalSum = 0;
        int n = s.length();
        
        for (int i = 0; i < n; i++) {
            // Get character at current 0-indexed position
            char c = s.charAt(i);
            
            // Calculate reversed alphabet position ('a' -> 26, 'b' -> 25, etc.)
            int revVal = 26 - (c - 'a');
            
            // Problem requires 1-indexed string positions, so multiply by (i + 1)
            totalSum += revVal * (i + 1);
        }
        
        return totalSum;
    }
}

File: 3811-reverse-degree-of-a-string/reverse-degree-of-a-string.py
class Solution(object):
    def reverseDegree(self, s):
        """
        :type s: str
        :rtype: int
        """
        total_sum = 0
        
        # enumerate(s, 1) starts the character index at 1 instead of 0
        for i, char in enumerate(s, 1):
            # Calculate reversed alphabet position ('a' -> 26, 'b' -> 25, etc.)
            rev_val = 26 - (ord(char) - ord('a'))
            
            # Multiply the value by its position in the string and add to total
            total_sum += rev_val * i
            
        return total_sum
